src/mpi/datatype/typerep/dataloop/dataloop_create_contig.c

```diff
@@ -44,7 +44,7 @@ int MPIR_Dataloop_create_contiguous(MPI_Aint icount, MPI_Datatype oldtype, void
 
         /* if we have a simple combination of contigs, coalesce */
         if (((old_loop_ptr->kind & MPII_DATALOOP_KIND_MASK) == MPII_DATALOOP_KIND_CONTIG)
-            && (old_size == old_extent)) {
+            && (old_size == old_extent) && (old_loop_ptr->el_size == old_loop_ptr->el_extent)) {
```

Comment:
Do we still need the `old_size == old_extent` check?  I think the `old_loop_ptr->el_size == old_loop_ptr->el_extent` check already covers that.

Reply:
The `el_size` and `el_extent` just makes sure the element is not resized. The datatype still can be resized even when the elment is not.

Reply:
There's already a check for `CONTIG`.  For contig types, when the element size is the same as extent, the total datatype size should also be equal to the total extent.  That's why I think the extra check is redundant.

Reply:
A "resized" type also poses as a `CONTIG` dataloop. In fact, quite a few dataloop optimization end up as a `CONTIG` dataloop, so any "should" is not given 😃

Reply:
Yes, I understand.  That's why the check for the element sizes was added.  Do you have a specific case where the extra check for the `old_size == old_extent` is needed beyond the element size check?

Reply:
The solaris failure case #4483, when the type is 
```
0: blkindx [numblks 1, blklen 131072, displs (0)]
1: resized [lb -400, extent 4]
2: resized [lb 0, extent 40]
```
The `blkindx` is optimized as a contig dataloop, and two resized type become a contig type with resized single element but `size==extent` due to `1: resized...` resized the extent back to `size` ... it is a corner case that is very difficult to randomly reproduce.

Reply:
I don't understand how the element size check is not sufficient in this case.

Reply:
In this case, the element size check is sufficient because it is the element size disagree. For general single resized type, the element size will agree but it will be the type size doesn't agree.

Reply:
For example, if we reverse the two `resized` type, the condition switches

Reply:
Do you have a specific case where the extra check for the `old_size == old_extent` is needed beyond the element size check?

Reply:
I think the dominant cases will fail if we remove that check.

Let me push a test commit and test it ...